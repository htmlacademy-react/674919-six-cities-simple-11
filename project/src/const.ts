export enum AppRoute {
  Main = '/',
  Login = '/login',
  Offer= '/offer/:id',
  PageNotExist= '/404',
}

export const PageModifierClassType: {[key: string]: string} = {
  [AppRoute.Main]: 'page--gray page--main',
  [AppRoute.Login]: 'page--gray page--login',
} as const;

export enum AuthorizationStatus {
  Auth = 'AUTH',
  NoAuth = 'NO_AUTH',
  Unknown = 'UNKNOWN',
}

export enum ReviewStatus {
  ReviewRest = 'REVIEW_REST',
  ReviewPending = 'REVIEW_PENDING',
  ReviewFulfilled = 'REVIEW_FULFILLED',
}

export const Ratings: string[] = ['20%', '40%', '60%', '80%', '100%'];

export const FiveStar: number[] = [5, 4, 3, 2, 1];
export const Months: string[] = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];

export enum Markers {
  Default = '/img/pin.svg',
  Active = '/img/pin-active.svg',
}

export enum City {
  Title = 'Amsterdam',
  Lat = 52.37454,
  Lng = 4.897976,
  Zoom = 13,
}

export enum PropertyClassName {
  MapPageOffer = 'property__map',
  MapPageMain = 'cities__map',
  PlaceCardListNear = 'near-places__list',
  PlaceCardItemNear = 'near-places',
  PlaceCardListCities = 'cities__places-list tabs__content',
  PlaceCardItemCities = 'cities',
}

export enum CitiesType {
  PARIS = 'Paris',
  COLOGNE = 'Cologne',
  BRUSSELS = 'Brussels',
  AMSTERDAM = 'Amsterdam',
  HAMBURG = 'Hamburg',
  DUSSELDORF = 'Dusseldorf',
}

// export const CitiesCoordinates: {[key: string]: string} = {
//   [CitiesType.PARIS]: [48.853, 2.348],
//   [CitiesType.COLOGNE]: [48.853, 2.348],
//   [CitiesType.BRUSSELS]: [48.853, 2.348],
//   [CitiesType.AMSTERDAM]: [48.853, 2.348],
//   [CitiesType.HAMBURG]: [48.853, 2.348],
//   [CitiesType.DUSSELDORF]: [48.853, 2.348],
// } as const;

export const Cities: string[] = ['Paris', 'Cologne', 'Brussels', 'Amsterdam', 'Hamburg', 'Dusseldorf'];

export const SortList = {
  POPULAR: 'Popular',
  PRICE_LOW: 'Price: low to high',
  PRICE_HIGH: 'Price: high to low',
  TOP_RATED_FIRST: 'Top rated first',
} as const;

export enum APIRoute {
  Offers = '/hotels',
  Offer = '/hotels/:id',
  Login = '/login',
  Logout = '/logout',
  Comments = '/comments',
}

export const TIMEOUT_SHOW_ERROR = 4000;
