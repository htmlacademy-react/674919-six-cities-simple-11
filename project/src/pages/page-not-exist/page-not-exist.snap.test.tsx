import {render} from '@testing-library/react';
import {createMemoryHistory} from 'history';
import PageNotExist from './page-not-exist';
import HistoryRoute from '../../components/history-route/history-route';
import {Provider} from 'react-redux';
import {configureMockStore} from '@jedmao/redux-mock-store';
import {AuthorizationStatus} from '../../const';

describe('Component: NotFoundScreen', () => {
  const mockStore = configureMockStore();

  it('should render correctly', () => {
    const store = mockStore({
      USER: {authorizationStatus: AuthorizationStatus.NoAuth},
    });
    const history = createMemoryHistory();

    const {container} = render(
      <Provider store={store}>
        <HistoryRoute history={history}>
          <PageNotExist/>
        </HistoryRoute>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
