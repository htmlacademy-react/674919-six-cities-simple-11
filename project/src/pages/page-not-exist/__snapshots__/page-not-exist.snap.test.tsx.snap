// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: NotFoundScreen should render correctly 1`] = `
<div>
  <div
    class="page page--gray"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link"
              href="/"
              title="Перейти на главную"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item"
              >
                <a
                  class="header__nav-link"
                  href="/login"
                >
                  <span
                    class="header__signout"
                  >
                    Sign in
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--property"
    >
      <div
        class="container"
      >
        404 Not Found
      </div>
    </main>
  </div>
</div>
`;
